dahdi_dummy: Build failure with kernel >= 6.15.3

init_module(): Magic init_module() function name is deprecated, use module_init(fn) instead.

see
https://lore.kernel.org/all/366bfdbe92736cde9fb01d5d3eb9b98e9070a1ec.1742852846.git.jpoimboe@kernel.org/
https://git.kernel.org/pub/scm/linux/kernel/git/tip/tip.git/commit/?id=4fab2d7628dd38f3fa8a5e615199350ecaeb17a8


build error

```
dahdi_dummy.o: error: objtool: init_module(): Magic init_module() function name is deprecated, use module_init(fn) instead
make[4]: *** [/usr/src/linux-6.15.5-1/scripts/Makefile.build:204: dahdi_dummy.o] Error 255
```

--- a/drivers/dahdi/dahdi_dummy.c
+++ b/drivers/dahdi/dahdi_dummy.c
@@ -209,7 +209,7 @@ static int dahdi_dummy_initialize(struct dahdi_dummy *ztd)
 	return res;
 }
 
-int init_module(void)
+static int __init dummy_init(void)
 {
 	int res;
 	ztd = kzalloc(sizeof(*ztd), GFP_KERNEL);
@@ -250,8 +250,7 @@ int init_module(void)
 	return 0;
 }
 
-
-void cleanup_module(void)
+static void __exit dummy_exit(void)
 {
 #if defined(USE_HIGHRESTIMER)
 	/* Stop high resolution timer */
@@ -272,3 +271,6 @@ module_param(debug, int, 0600);
 MODULE_DESCRIPTION("Timing-Only Driver");
 MODULE_AUTHOR("Robert Pleh <robert.pleh@hermes.si>");
 MODULE_LICENSE("GPL v2");
+
+module_init(dummy_init);
+module_exit(dummy_exit);

